test/e2e/apps/replica_set.go

```diff
@@ -524,3 +534,138 @@ func testRSLifeCycle(f *framework.Framework) {
 
 	framework.ExpectNoError(err, "failed to see replicas of %v in namespace %v scale to requested amount of %v", rs.Name, ns, rsPatchReplicas)
 }
+
+func testRSStatus(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+	rsClient := c.AppsV1().ReplicaSets(ns)
+
+	// Define ReplicaSet Labels
+	rsPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  WebserverImageName,
+	}
+	labelSelector := labels.SelectorFromSet(rsPodLabels).String()
+	framework.Logf("labels: %v", labelSelector)
+
+	rsName := "test-rs"
+	replicas := int32(1)
+
+	w := &cache.ListWatch{
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.LabelSelector = labelSelector
+			return rsClient.Watch(context.TODO(), options)
+		},
+	}
+	rsList, err := c.AppsV1().ReplicaSets("").List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+	framework.ExpectNoError(err, "failed to list Replicasets")
+
+	ginkgo.By("Create a Replicaset")
+	rs := newRS(rsName, replicas, rsPodLabels, WebserverImageName, WebserverImage, nil)
+	testReplicaSet, err := c.AppsV1().ReplicaSets(ns).Create(context.TODO(), rs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("Verify that the required pods have come up.")
+	err = e2epod.VerifyPodsRunning(c, ns, "sample-pod", false, replicas)
+	framework.ExpectNoError(err, "Failed to create pods: %s", err)
+
+	ginkgo.By("Getting /status")
```

Comment:
Is the goal here just to hit `/status` endpoint? I ask because to get status sub-resource, `.Get()` method should have been enough similar to the approach you followed below.

Reply:
@ravisantoshgudimetla I'm on the same team has @Riaankl and wrote this test to be part of conformance.

Yes, the goal is to hit the `/status` endpoint. The `.Get()` method can't access the status sub-resource, so it requires the dynamic client.